Trainer_finetune.py

```diff
@@ -30,18 +36,29 @@ def device(self):
         self.net.to(torch.device("cuda"))
 
     def load_model(self, name=None, rank=0, real=False):
-        def convert(param):
-            return {
-            k.replace("module.", ""): v
-                for k, v in param.items()
-                if "module." in k and 'attn_mask' not in k and 'HW' not in k
-            }
         if rank <= 0 :
             if name is None:
                 name = self.name
             print(f"loading {name} ckpt")
             self.net.load_state_dict(convert(torch.load(f'ckpt/{name}.pkl')), strict=True)
 
+    def from_pretrained(self, model_name):
+        try:
+            from huggingface_hub import hf_hub_download
+            assert model_name in ["VFIMamba", "VFIMamba_S"], "Please select a valid model name from ['VFIMamba', 'VFIMamba_S']"
+
+            ckpt_path = hf_hub_download(
+                repo_id=f"MCG-NJU/{model_name}", filename=model_name + ".pkl"
+            )
+            checkpoint = torch.load(ckpt_path)
+        except:
+            # In case the model is not hosted on huggingface
+            # or the user cannot import huggingface_hub correctly, model_name option: VFIMamba, VFIMamba_S
+            _VFIMAMBA_URL = f"https://huggingface.co/MCG-NJU/{model_name}/resolve/main/{model_name}.pkl"
+            checkpoint = torch.hub.load_state_dict_from_url(_VFIMAMBA_URL)
+
+        self.net.load_state_dict(convert(checkpoint), strict=True)

```

Comment:
```suggestion
    @classmethod
    def from_pretrained(cls, model_id: str, local_rank: int) -> "Model":
        try:
            from huggingface_hub import hf_hub_download
        except ImportError:
            raise ImportError(
                "Model is hosted on the Hugging Face Hub. "
                "Please install huggingface_hub by running `pip install huggingface_hub` to load the weights correctly."
            )

        if "/" not in model_id:
            model_id = "MCG-NJU/" + model_id

        ckpt_path = hf_hub_download(repo_id=model_id, filename="model.pkl")
        checkpoint = torch.load(ckpt_path)
        
        model = cls(local_rank)
        model.net.load_state_dict(convert(checkpoint), strict=True)
        return model

```

Hi there, maintainer of `huggingface_hub` here :wave: May I suggest a more opinionated implementation for this method? In particular:
- I would make `huggingface_hub` required and raise an explicit error if it's not the case. This way you don't have to maintain different ways to load the model. In particular, end users will benefit from the `huggingface_hub` cache when running the script several times.
- Now that models are in two separate repos, I would rename the weights files to `model.pkl` in both cases. This way, the repo id and filename don't have to be correlated. Also, it would simplify the way downloads are counted.
- I would also accept any `model_id` as input, not only the 2 official one from the repo. This way, your library can be reused by anyone wanting the train, retrain, fine-tuned, etc new models based on your work. This should greatly improve adoption of the library. `model_id` can be either a full repo id (e.g. `MCG-NJU/VFIMamba_S`) or simply a model name in which case the `MCG-NJU/` org is prefixed.
- I would make `from_pretrained` a class method to initialize the object directly with `model = Model.from_pretrained("VFIMamba_S", local_rank=-1)` 


Those changes are not an obligation to make things work with the Hub but in my opinion it would greatly improve the integration. Let me know what you think!

(cc @hanouticelina for viz')

Reply:
Thank you very much! Your suggestion is very helpful. I've made some changes based on your advice.